grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandler.java

```diff
@@ -102,11 +105,17 @@ static UnframedGrpcErrorHandler ofJson(UnframedGrpcStatusMappingFunction statusM
                 cause = null;
             }
             final HttpStatus httpStatus = mappingFunction.apply(ctx, status, cause);
-            final ResponseHeaders responseHeaders = ResponseHeaders.builder(httpStatus)
-                                                                   .contentType(MediaType.JSON_UTF_8)
-                                                                   .addInt(GrpcHeaderNames.GRPC_STATUS,
-                                                                           grpcCode.value())
-                                                                   .build();
+            final ResponseHeadersBuilder responseHeadersBuilder =
+                    ResponseHeaders.builder(httpStatus)
+                                   .contentType(MediaType.JSON_UTF_8)
+                                   .addInt(GrpcHeaderNames.GRPC_STATUS,
+                                           grpcCode.value());
+            final HttpHeaders trailers = !response.trailers().isEmpty() ?
+                                         response.trailers() : response.headers();
+            final String grpcStatusDetailsBin = trailers.get(GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN);
+            if (!Strings.isNullOrEmpty(grpcStatusDetailsBin)) {
+                responseHeadersBuilder.set(GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN, grpcStatusDetailsBin);
+            }
```

Comment:
Shouldn't we need to render a rich error response using the `GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN`?
https://cloud.google.com/apis/design/errors#error_model
I guess users don't want to deserialize Protobuf data in headers.
https://cloud.google.com/apis/design/errors#error_model
```json
{
  "error": {
    "code": 400,
    "message": "API key not valid. Please pass a valid API key.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "API_KEY_INVALID",
        "domain": "googleapis.com",
        "metadata": {
          "service": "translate.googleapis.com"
        }
      }
    ]
  }
}
```

Reply:
On top of that, is it better to modify the default error response format for better compatibility with google APIs?
For example:
* if `GRPC_STATUS_DETAILS_BIN is not specified
```json
{
  "error": {
    "code": 400,
    "message": "API key not valid. Please pass a valid API key.",
    "status": "INVALID_ARGUMENT"
  }
}
```
* if `GRPC_STATUS_DETAILS_BIN is specified
```json
{
  "error": {
    "code": 400,
    "message": "API key not valid. Please pass a valid API key.",
    "status": "INVALID_ARGUMENT",
    "details": [ ... ]
  }
}
```

Reply:
Oh, sure. I originally planned to include the rich error model in my next PR. But yes, I would finish that later on in this PR. Thanks for the advise.

Reply:
Do you think is it useful to include `grpc-status-detail-bin` in response headers? I thought we don't need to propagate the raw binary header.

Reply:
If you are going to handle the rich error model, how about reverting the changes in `UnframedGrpcErrorHandler` and handling it together in the next PR?

Reply:
> If you are going to handle the rich error model, how about reverting the changes in `UnframedGrpcErrorHandler` and handling it together in the next PR?

Oh, sure. I will think about whether to include grpc-status-detail-bin in response headers later on. And can you advise me on how to write a test case if I do not include that in response headers for this PR?

Reply:
I think we can capture a `ResponseHeaders` by adding a server-slide decorator.

Reply:
@ikhoon  reverted.